[CODEOWNERS] Update agent-shared-components components with correct ownership

sgnn7 opened the pull request:
### What does this PR do?

Changes ownership of some files and directories from @DataDog/agent-core
to @DataDog/agent-shared-components.

### Motivation

Since @DataDog/agent-core is split now, we have some clarity on the
components that will be owned by @DataDog/agent-shared-components. 

### Additional Notes

We still have a few items that will need sorting but this should be enough
to get us started.

### Possible Drawbacks / Trade-offs

N/A

### Describe how to test/QA your changes

N/A

### Reviewer's Checklist

- [ ] If known, an appropriate milestone has been selected; otherwise the `Triage` milestone is set.
- [ ] Use the `major_change` label if your change either has a major impact on the code base, is impacting multiple teams or is changing important well-established internals of the Agent. This label will be use during QA to make sure each team pay extra attention to the changed behavior. For any customer facing change use a releasenote.
- [ ] A [release note](https://github.com/DataDog/datadog-agent/blob/main/docs/dev/contributing.md#reno) has been added or the `changelog/no-changelog` label has been applied.
- [ ] Changed code has automated tests for its functionality.
- [ ] Adequate QA/testing plan information is provided if the `qa/skip-qa` label is not applied.
- [ ] At least one `team/..` label has been applied, indicating the team(s) that should QA this change.
- [ ] If applicable, docs team has been notified or [an issue has been opened on the documentation repo](https://github.com/DataDog/documentation/issues/new).
- [ ] If applicable, the `need-change/operator` and `need-change/helm` labels have been applied.
- [ ] If applicable, the [config template](https://github.com/DataDog/datadog-agent/blob/main/pkg/config/config_template.yaml) has been updated.

sgnn7 committed:
[CODEOWNERS] Update agent-shared-components components with correct ownership

Since @DataDog/agent-core is split now, we have some clarity on the
components that will be owned by @DataDog/agent-shared-components. We
still have a few items that will need sorting but this should be enough
to get us started.

mx-psi reviewed (approved):
Slack map LGTM

carlosroman commented on .github/CODEOWNERS:
```diff
@@ -139,14 +139,14 @@
 /pkg/dogstatsd/                         @DataDog/agent-metrics-logs
 /pkg/forwarder/                         @DataDog/agent-metrics-logs
```
```suggestion
/pkg/forwarder/                         @DataDog/agent-shared-components
```
All looks good. There was a conversation that maybe this should become responsibility of @DataDog/agent-shared-components. I can see both sides to the argument, so we can revise this at another date.

sgnn7 replied:
Indeed - I based this on that conversation and the original notes on the split. I think the forwarder should be indeed part of ASC responsibility since it's more generally about sending data rather than processing metrics.

olivielpeau commented on .github/CODEOWNERS:
```diff
@@ -128,7 +128,7 @@
 /Makefile.trace                         @DataDog/agent-apm
 
 /omnibus/                               @DataDog/agent-platform
-/omnibus/config/software/datadog-agent*.rb                @Datadog/agent-core @DataDog/agent-platform
```
I think we should remove this line entirely, having `/omnibus/config/software/datadog-agent*.rb` owned by agent-platform like the rest of `/omnibus/` makes sense to me

olivielpeau reviewed (approved):
LGTM!

Also, L136: the catchall `/pkg/` should be owned by `agent-shared-components` (although my expectation is that if the CODEOWNERS file is exhaustive nothing should be matched by this line, I'd expect all packages in `/pkg/` to be matched by a more precise rule)

All the remaining references to `agent-core` should IMO either be removed entirely, or owned by `agent-metrics-logs`.

sgnn7 committed:
[CODEOWNERS] Update the PR based on feedback

We needed to replace more instances of unclear or mixed ownership to
clearly be owned by a specific team.

olivielpeau commented on .github/CODEOWNERS:
```diff
@@ -182,9 +183,20 @@
 /pkg/remoteconfig/                      @DataDog/remote-config
+/pkg/runtime/                           @DataDog/agent-shared-components
+/pkg/secrets/                           @DataDog/agent-shared-components
+/pkg/serializer/                        @DataDog/agent-shared-components
```
To me, this is tied to the aggregator, and more generally to metrics and logs (at least at the moment) so I'd define metrics-logs as the owner:

```suggestion
/pkg/serializer/                        @DataDog/agent-metrics-logs
```

olivielpeau commented on .github/CODEOWNERS:
```diff
@@ -231,18 +243,18 @@
 /releasenotes-dca/                      @DataDog/documentation @DataDog/container-integrations @Datadog/container-app
 
-/rtloader/                              @DataDog/agent-core
+/rtloader/                              @DataDog/agent-shared-components
```
Since it's tied to the collector and python checks

```suggestion
/rtloader/                              @DataDog/agent-metrics-logs
```

olivielpeau reviewed (approved):
Just a few suggestions

sgnn7 committed:
Update .github/CODEOWNERS

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

sgnn7 committed:
Update .github/CODEOWNERS

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>

sgnn7 committed:
Update .github/CODEOWNERS

Co-authored-by: Olivier Vielpeau <olivielpeau@users.noreply.github.com>
